### PR TITLE
Implemented [meta.trans.other] `aligned_union` for libstdc++-4.8

### PR DESCRIPTION
**Pull request by K-ballo**

### CODE REVIEW

File: include/slb/type_traits.hpp [hunk 1]
```diff
@@ -903,11 +903,63 @@ using add_pointer_t = typename slb::add_pointer<T>::type;
...
+#endif
+#else
+#define SLB_CPP_LIB_ALIGNED_UNION 0 // not available
+#endif
+
+#if SLB_CPP_LIB_ALIGNED_UNION == 2
+using std::aligned_union;
+#else
+#if SLB_CPP_LIB_ALIGNED_UNION == 1
+template <std::size_t Len, typename... Types>
```
Comment (mpark) [inline include/slb/type_traits.hpp]: Personally I prefer `Ts`

File: include/slb/type_traits.hpp [hunk 2]
```diff
@@ -903,11 +903,63 @@ using add_pointer_t = typename slb::add_pointer<T>::type;
...
+struct aligned_union : std::aligned_union<Len, Types...> {
+  static constexpr std::size_t alignment_value =
+      std::aligned_union<Len, Types...>::alignment_value;
+};
+#else
+namespace detail {
+template <std::size_t... Vs>
+struct static_max {
```
Comment (mpark) [inline include/slb/type_traits.hpp]: Can we leave this one undefined?

File: include/slb/type_traits.hpp [hunk 3]
```diff
@@ -903,11 +903,63 @@ using add_pointer_t = typename slb::add_pointer<T>::type;
...
+  static constexpr std::size_t value = 0u;
+};
+
+template <std::size_t V0>
+struct static_max<V0> {
```
Comment (mpark) [inline include/slb/type_traits.hpp]: I think calling this `V0` is confusing seeing as it's not the same `V0` from the recursive case.
Can we just call it `V`?

File: test/type_traits.cpp [hunk 4]
```diff
@@ -631,6 +631,38 @@ TEST_CASE("add_pointer_t", "[meta.trans.ptr]") {
...
+    CHECK(std::is_trivial<type>::value);
+    CHECK(std::is_standard_layout<type>::value);
+    CHECK(alignof(type) % alignof(U) == 0);
+
+    CHECK(sizeof(slb::aligned_union<sizeof(int) * 2, int>::type) >=
+          sizeof(int) * 2);
+  }
+
+  /* The static member alignment_value shall be an integral constant of type
+  size_t whose value is the strictest alignment of all types listed in Types.*/
```
Comment (mpark) [inline test/type_traits.cpp]: These comments are formatted weird :( probably because ClangFormat tries not to touch comments.

**Review by mpark — APPROVED** [review]

File: test/type_traits.cpp (hunk 4 above)
Comment (K-ballo) [reply test/type_traits.cpp]: How should they be formatted instead?

Comment (mpark) [reply test/type_traits.cpp]: I think for a paragraph like this maybe we should just do:
```
// The static member alignment_value shall be an integral constant of type
// size_t whose value is the strictest alignment of all types listed in Types.
{
  ...
}
```

File: include/slb/type_traits.hpp [hunk 5]
```diff
@@ -903,11 +903,63 @@ using add_pointer_t = typename slb::add_pointer<T>::type;
 // 23.15.7.6, other transformations
 
 using std::aligned_storage;
-using std::aligned_union;
 using std::decay;
 using std::enable_if;
 using std::conditional;
 
+#if !defined(__GLIBCXX__) || __has_include(<codecvt>) // >= libstdc++-5
```
Comment (mpark) [inline include/slb/type_traits.hpp]: One more thing: can we move this whole block up to where `aligned_union` was before?

Comment (K-ballo) [reply include/slb/type_traits.hpp]: Certainly! I thought we were putting pieces that we override at the bottom of their category.

**Pull request merged**